features/192-asset-gear/index.md

```diff
@@ -0,0 +1,495 @@
+```
+
+Transform the asset into JPEG.
+
+##### quality
+
+Set the quality if the format is `jpg` or `webp`.
+
+- `Q`: Set the absolute quality. Valid values are 1 to 100.
+
+###### Example
+
+```
+pipeline=image/format,jpg/quality,Q_85
+```
+
+Transform the asset into JPEG of quality 0.85.
+
+###### resize
+
+- `m`: Specify the scaling mode. Valid values are `lfit`, `mfit`, `fill`, `pad` and `fixed`.
+- `w`: Specify the target width. Valid values are 1 to 4096.
+- `h`: Specify the target height. Valid values are 1 to 4096.
+- `l`: Specify the longer side of the target. Valid values are 1 to 4096.
+- `s`: Specify the shorter side of the target. Valid values are 1 to 4096.
+- `color`: Specify the fill color when `m` is `pad`.
+
+For the details on the scaling mode, See https://www.alibabacloud.com/help/doc-detail/44688.htm
+
+###### Example
+
+```
+pipeline=image/resize,m_fill,w_200,h_200
+```
+
+Crop the image into 200px x 200px.
+
+```
+pipeline=image/resize,l_200
+```
+
+Resize the image such that its longer side is 200px. Its shorted side is resized proportionally.
+
+#### Unsupported Operations
+
+##### Encoding into Progressive JPEG
+
+Encoding into progressive JPEG is not supported because it is not performant.
+See https://github.com/libvips/libvips/issues/77
+
+##### 8-bit PNG
+
+libvips does not support transform into 8-bit PNG without saving to a local file.
+
+### Image Processing via HTTP Client Hints
+
+To be specified in the future.
+
+Some useful references
+
+- https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/client-hints/#responsive_images
+
+#### Interaction between HTTP Client Hints and Query String
+
+To be specified in the future.
+
+#### Caveat
+
+The initial implementation does not have caching.
+
+## SDK API
+
+### Web
+
+```typescript
+interface UploadAssetOptions {
+  // access by default is `public`.
+  access?: "public" | "private",
+  // contentType is required when `blob.type` does not have value.
+  contentType?: string;
+}
+
+function upload(blob: Blob, options?: UploadAssetOptions): Promise<string>;
+```
+
+### Node Client
+
+```typescript
+interface UploadAssetOptions {
+  // access by default is `public`.
+  access?: "public" | "private",
+  // contentType is required because neither Buffer nor Readable has this information.
+  contentType: string;
+  // contentLength is required when data is Readable.
+  contentLength?: number;
+  // contentMD5 is required when data is Readable.
+  contentMD5?: string;
+}
+
+function upload(data: Buffer | stream.Readable, options: UploadAssetOptions): Promise<string>;
+```
+
+### React Native
+
+```typescript
+interface UploadAssetOptions {
+  // access by default is `public`.
+  access?: "public" | "private",
+  // contentType is required because the SDK does not support inferring media type
+  // from file extension.
+  contentType: string;
+}
+
+function upload(specifier: string, options: UploadAssetOptions): Promise<string>;
```

Comment:
Instead of using a new word for this param, maybe we can just call it `uri`, easier for user to understand.